src/text2.c

```diff
@@ -2463,11 +2467,746 @@ void sub_8007D00(UnkTextStruct1 *a0, u32 a1, s32 x, s32 y, s32 a4, s32 color)
     "\tbx r0");
 }
 
-void sub_8007E20(u32 a0, u32 a1, u32 a2, u32 a3, u32 a4, u8 *a5, u32 a6)
+void sub_8007E20(u32 a0, u32 a1, u32 a2, u32 a3, u32 a4, u32 *a5, u32 a6)
 {
-    sub_8007E64(gUnknown_2027370, &gUnknown_202B038[0][0][0], a0, a1, a2, a3, a4, a5, a6);
+    sub_8007E64(gUnknown_2027370, gUnknown_202B038[0], a0, a1, a2, a3, a4, a5, a6);
 }
 
 UNUSED static void nullsub_160(void)
 {
 }
+
+void sub_8007E64(UnkTextStruct1 *a0, u16 a1[32][32], u32 a2, s32 a3, s32 a4, s32 a5, s32 a6, u32 *a7, u32 a8)
+{
+    s32 i, j;
+    UnkTextStruct1 *strPtr = &a0[a2];
+
+    a3 /= 8;
+    a4 /= 8;
+    a5 /= 8;
+    a6 /= 8;
+    a8 *= 4096;
+    if (a4 < strPtr->unk8) {
+        s32 id = (strPtr->unk4 * a4) + a3;
+        u32 *unk18Ptr = &strPtr->unk18[id * 8];
+
+        for (i = 0; i < a6; i++) {
+            s32 xMaybe = a3;
+            u32 *loopUnk18Ptr = unk18Ptr;
+            for (j = 0; j < a5; j++) {
+                if (xMaybe < strPtr->unk4) {
+                    if (strPtr->unk3C > loopUnk18Ptr) {
+                        strPtr->unk3C = loopUnk18Ptr;
+                    }
+                    *(loopUnk18Ptr++) = *(a7++);
+                    *(loopUnk18Ptr++) = *(a7++);
+                    *(loopUnk18Ptr++) = *(a7++);
+                    *(loopUnk18Ptr++) = *(a7++);
+                    *(loopUnk18Ptr++) = *(a7++);
+                    *(loopUnk18Ptr++) = *(a7++);
+                    *(loopUnk18Ptr++) = *(a7++);
+                    *(loopUnk18Ptr++) = *a7;
+                    if (strPtr->unk40 < loopUnk18Ptr) {
+                        strPtr->unk40 = loopUnk18Ptr;
+                    }
+                    a7++;
+                    a1[strPtr->unk2 + a4][strPtr->unk0 + xMaybe] &= 0xFFF;
+                    a1[strPtr->unk2 + a4][strPtr->unk0 + xMaybe] |= a8;
+                }
+                else {
+                    loopUnk18Ptr += 8;
+                    a7 += 8;
+                }
+                xMaybe++;
+            }
+            a4++;
+            unk18Ptr += strPtr->unk20;
+            unk18Ptr += 8;
+            if (a4 >= strPtr->unk8)
+                break;
+        }
+    }
+}
+
+u32 sub_8007FA8(u32 a0)
```

Comment:
This reverses the nybbles in a u32, AKA flips pixels horizontally in a tile's row of data.
Probably worth renaming appropriately

Reply:
Something like FlipPixelsHorizontally works?

Reply:
Yep